hamilton/io/materialization.py

```diff
@@ -247,7 +247,9 @@ def join_function(**kwargs):
                 callabl=join_function,
                 input_types={dep.name: dep.type for dep in node_dependencies},
                 originating_functions=(
-                    None if self.result_builder is None else [self.result_builder.build_result]
```

Comment:
This will be a locally defined function and will has the same for every single case in which it is not `None`. Is that intended?

Reply:
I guess in this case that makes sense -- it should just be the same?

Reply:
reverting